### ISSUE TITLE
Highly Coupled System of PDE

### ISSUE THREAD

**Issue by ShupengGao** [lululxvi/deepxde#1319]
Hi,Lu
There is a highly coupled system of partial differential equations,I encountered the following issues:
1)The gradient decreases slowly and fluctuates greatly, even disappearing.
I tried resampling, PFNN, hard initial conditions, loss_weights,etc., but there was no improvement.Can you give me some suggestions?
The code is as follows:

def pde(x,y):
    GHe=function1(x[:,0:1],x[:,1:2])
    Gv=function2(x[:,0:1],x[:,1:2])
    Delta=b * Gv
    Cv,Ci,Cg,Cgv,C2gv,C2g,Cs,Cb,m1,Rb = y[:,0:1],y[:,1:2],y[:,2:3],y[:,3:4],y[:,4:5],y[:,5:6],y[:,6:7],y[:,7:8],y[:,8:9],y[:,9:10]
    Cv_t = dde.grad.jacobian(y, x, i=0, j=2)
    Cv_xx =dde.grad.hessian(y, x, component=0, i=0, j=0)
    Cv_yy = dde.grad.hessian(y, x, component=0, i=1, j=1)
    Ci_t = dde.grad.jacobian(y, x, i=1, j=2)
    Ci_xx = dde.grad.hessian(y, x, component=1, i=0, j=0)
    Ci_yy = dde.grad.hessian(y, x, component=1, i=1, j=1)
    Cg_t = dde.grad.jacobian(y, x, i=2, j=2)
    Cg_xx = dde.grad.hessian(y, x, component=2, i=0, j=0)
    Cg_yy = dde.grad.hessian(y, x, component=2, i=1, j=1)
    Cgv_t = dde.grad.jacobian(y, x, i=3, j=2)
    C2gv_t = dde.grad.jacobian(y, x, i=4, j=2)
    C2g_t = dde.grad.jacobian(y, x, i=5, j=2)
    Cs_t = dde.grad.jacobian(y, x, i=6, j=2)
    Cb_t = dde.grad.jacobian(y, x, i=7, j=2)
    m1_t = dde.grad.jacobian(y, x, i=8, j=2)
    Rb_t = dde.grad.jacobian(y, x, i=9, j=2)
    Epsilon = (4 * np.pi / 48) * (Rb / a0) * 10 ** 3 / Time
    Csv = (a0 ** 2 / 48) * (Zv * p + 4 * np.pi * Rb * Cb / omega)
    Csi = (a0 ** 2 / 48) * (Zi * p + 4 * np.pi * Rb * Cb / omega)

    equ1 = Dv * (Cv_xx + Cv_yy) + f * Gv + (Delta + Beta * e1) * Cgv - (Alpha * Ci + Beta * Cg + Gamma * (Csv + Cgv + 2 * C2g + 2 * C2gv + 3 * Cs)) * Cv - Cv_t
    equ2 = Di * (Ci_xx + Ci_yy) + f * Gv - Alpha * Ci * (Cv + Cgv + 2 * C2gv + 3 * Cs + Csi) - Ci_t
    equ3 = Dg * (Cg_xx + Cg_yy) + GHe + Beta * (e1 * Cgv + e2 * C2gv + 2 * e3 * C2g) + Delta * (Cgv + 2 * C2gv + 4 * C2g + 3 * Cs + m1 * Cb) + Alpha * Ci * (Cgv + 3 * Cs) - Beta * Cg *(Cv + 4* Cg + Cgv + 2 * C2gv + 2 * C2g + 3 * Cs + Epsilon * Cb) - Cg_t
    equ4 = Beta * e2 * C2gv + 2 * Delta * C2gv + Beta * Cg * Cv + 2 * Gamma * Cv * C2gv - Cgv * (Beta * e1 + Delta + Alpha * Ci + Beta * Cg) - Cgv_t
    equ5 = 3 * Delta * Cs + Beta * Cg * Cv + 2 * Gamma * Cv * C2g - C2gv * (Beta * e2 + 2 * Delta + 2 * Beta * Cg + 2 * Alpha * Ci + 2 * Gamma * Cv) - C2gv_t
    equ6 = 2 * Alpha * Ci * C2gv + 2 * Beta * Cg ** 2 - C2g * (Beta * e3 + 2 * Delta + 2 * Gamma * Cv + 2 * Beta * Cg) - C2g_t
    equ7 = 2 * Beta * Cg * (C2gv + C2g) - 3 * Cs * (Delta + Alpha * Ci + Beta * Cg + Gamma * Cv) - Cs_t
    equ8 = 1 / m1 * (12 * Beta * Cg * Cs + 9 * Gamma * Cv * Cs) - Cb_t
    equ9 = Epsilon * Beta * Cg - m1_t
    equ10 = 1 / Rb * (Dv * Cv - Di * Ci) - Rb_t
  
    return [equ1, equ2, equ3, equ4, equ5, equ6, equ7, equ8, equ9, equ10]

def feature_transform(x):
        return torch.cat(
            [
                x[:,0:1] * 0.01,
                x[:,1:2] * 0.01,
                x[:,2:3],
            ],
            dim=1,
        )
def output_transform(x,y):
    return torch.cat(
        [
            x[:,2:3] * y[:,0:1],
            x[:,2:3] * y[:,1:2],
            x[:,2:3] * y[:,2:3],
            x[:,2:3] * y[:,3:4],
            x[:,2:3] * y[:,4:5],
            x[:,2:3] * y[:,5:6],
            x[:,2:3] * y[:,6:7],
            x[:,2:3] * y[:,7:8],
            x[:,2:3] * y[:,8:9] + 3,
            x[:,2:3] * y[:,9:10] + 0.3,
        ],
        dim=1,
    )

geom = dde.geometry.PointCloud(inside, boundary_points=boundary)
timedomain = dde.geometry.TimeDomain(0,2)
geomtime = dde.geometry.GeometryXTime(geom, timedomain)

data = dde.data.TimePDE(
    geomtime,
    pde,
    [],
    num_domain=30000,
)
layer_size = [3] + [64] * 5 + [10]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.nn.FNN(layer_size, activation, initializer)
net.apply_feature_transform(feature_transform)
net.apply_output_transform(output_transform)
model = dde.Model(data, net)
model.compile("adam", lr=0.001, loss="MSE",loss_weights=[1,1,1,1,1,1,1,1000,1000,1])
pde_resampler = dde.callbacks.PDEPointResampler(period=1000)
cp_callback = dde.callbacks.ModelCheckpoint(filepath="function/model",verbose=0,save_better_only=True)
losshistory, train_state = model.train(iterations=50000,callbacks=[cp_callback,pde_resampler])
dde.saveplot(losshistory, train_state, issave=True, isplot=True)
